php/routes.php

```diff
@@ -26,4 +26,42 @@
             ]
         );
     });
+
+    CBPRouter::code('/code', function ($payload) {
+        $parsedUrl = wp_parse_url($payload['url']);
+
+        $isValidHost = $parsedUrl['host'] === 'gist.githubusercontent.com' ||
+            $parsedUrl['host'] === 'raw.githubusercontent.com' ||
+            $parsedUrl['host'] === 'github.com';
+
+        if (!$isValidHost) {
+            return new WP_REST_Response(
+                [
+                    'code' => 'Error: Invalid Host'
+                ]
+            );
+        }
```

Comment:
Should return the proper status code here, and maybe use message instead of code

```php
if (!$isValidHost) {
    return new WP_REST_Response([
        'message' => 'Invalid URL'
    ], 422);
}
```

If the text is later displayed to the user, it should also go in `__()` so it can be translated